New gui vehicle origin

amagaz24 opened the pull request:
Added option to right-click context menu that launches a popup with options to change the takeoff/land home for all or selected vehicles if they are currently airborne.

amagaz24 committed:
New Vehicle Home menu

amagaz24 committed:
Set Vehicle Home Functionality

amagaz24 committed:
Cleaning up unused lines

amagaz24 committed:
Enabling altitude transform

amagaz24 committed:
Solving MSL altitude interpretation

amagaz24 committed:
Simpler reference altitude correction

amagaz24 committed:
Simpler reference altitude correction

amagaz24 committed:
Build fix

amagaz24 committed:
Disable SetVehicleHome for grounded vehicles

patnolan33 commented on MACE_Frontend/src/components/map/components/environment-icon/index.tsx:
```diff
@@ -15,7 +15,11 @@ export default (props: Props) => {
     case "mission_target":
       return <MissionTarget {...props} />;
     case "takeoff_land":
+      if(props.data.name != "0"){
```
It looks like you are using the `name` field as a proxy for vehicle ID. If that's the case, I don't see a reason to "name" the vehicle home position, so just switch the `name` field to `agentID` if it doesn't already have one.

patnolan33 replied:
Name field is being used as a proxy for agent ID right now for home positions.

patnolan33 commented on MACE_Frontend/src/components/map/index.tsx:
```diff
@@ -65,29 +73,78 @@ export default class MapView extends React.Component<Props, State> {
+    let hide = false;
+    options.push(<option key={0} value="ALL">All Aircraft</option>);
+    this.props.context.aircrafts.forEach((a) => {
+      if (["STABILIZE",""].indexOf(a.mode) != -1){
```
Why are you checking for `STABILIZE`? I think the dropdown should just display all aircraft.

patnolan33 commented on MACE_Frontend/src/components/map/index.tsx:
```diff
@@ -65,29 +73,78 @@ export default class MapView extends React.Component<Props, State> {
+    let payloadArray = [];
+    let enabledAircraft = [];
+    this.props.context.aircrafts.forEach((a) => {
+      if (["STABILIZE",""].indexOf(a.mode) == -1){
```
Similar question to a previous comment: why check for the `STABILIZE` mode? I think if "ALL" is selected, then the home should be sent for every vehicle, not just those in `STABILIZE` mode. If specific modes are required for setting home positions, I think we would have to go one step further and actually require a "GROUNDED" state, for which we would need the MACE state machine status to properly handle.

patnolan33 commented on MACE_Frontend/src/components/map/index.tsx:
```diff
@@ -123,13 +180,65 @@ export default class MapView extends React.Component<Props, State> {
+          >
+            <span style={styles.inputLabel}>Select Aircraft:</span>
+            <div>
+              <select style={styles.selectRow} onChange={this.onAircraftSelected} >
```
Did you try using a `<Select />` component from `react-select`? This could allow for multiple selection options if we wanted to send a subset of vehicles a new home location. I think it would look something like this: 

```
<Select
    options={this.createDropdownList}
    isMulti
    onChange={(ev) => {
        this.onAddSelectedAircraft(ev);
    }}
/>
```
where `onAddSelectedAircraft` would simply append the aircraft to a list. You would have to handle the removal as well.

patnolan33 committed:
Added comment to guitomace 

We need to determine a more robust way to handle vehicle home altitude conversions. Our altitude reference frames should be handling this, however setting the home position did not result in correct relative altitude reporting.

patnolan33 commented on src/module_ground_station/macetogui.cpp:
```diff
@@ -74,7 +74,7 @@ void MACEtoGUI::sendVehicleHome(const int &vehicleID, const command_item::Spatia
     {
         QJsonObject obj = home.toJSON(0, guiMessageString(GuiMessageTypes::VEHICLE_HOME));
-        obj["name"] = QString::fromStdString("Agent " + std::to_string(vehicleID));
+        obj["name"] = QString::fromStdString(std::to_string(vehicleID));
```
Similar to comment above: if "name" is only acting as a vehicle ID, then just change this field name accordingly.

patnolan33 replied:
Name field is being used as a proxy for agent ID right now for home positions.

patnolan33 reviewed (changes requested):
Some clarification is needed, see comments.

amagaz24 committed:
Merge branch 'newGUI' into newGUI-vehicleOrigin
